testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OIDCProtocolMappersTest.java

```diff
@@ -784,6 +785,45 @@ public void testUserGroupRoleToAttributeMappersScopedClientNotSet() throws Excep
         deleteMappers(protocolMappers);
     }
 
+    @Test
+    public void testUserGroupRoleToAttributeMappersScopedWithDifferentClient() throws Exception {
+        String clientId = "test-app-scope";
+        String diffClient = "test-app";
+
+        ProtocolMapperRepresentation realmMapper = ProtocolMapperUtil.createUserRealmRoleMappingMapper("pref.", "Realm roles mapper", "roles-custom.realm", true, true);
+        ProtocolMapperRepresentation clientMapper = ProtocolMapperUtil.createUserClientRoleMappingMapper(diffClient, null, "Client roles mapper", "roles-custom.test-app", true, true);
+
+        ClientResource clientResource = ApiUtil.findClientResourceByClientId(adminClient.realm("test"), clientId);
+        assertThat(clientResource, notNullValue());
+
+        ProtocolMappersResource protocolMappers = clientResource.getProtocolMappers();
+        protocolMappers.createMapper(Arrays.asList(realmMapper, clientMapper));
+
+        // Login user
+        ClientManager.realm(adminClient.realm("test")).clientId(clientId).directAccessGrant(true);
+        oauth.clientId(clientId);
+        OAuthClient.AccessTokenResponse response = browserLogin("password", "rich.roles@redhat.com", "password");
+        IDToken idToken = oauth.verifyIDToken(response.getIdToken());
+
+        // Verify attribute is filled
+        Map<String, Object> roleMappings = (Map<String, Object>) idToken.getOtherClaims().get("roles-custom");
+        Assert.assertThat(roleMappings.keySet(), containsInAnyOrder("realm", diffClient));
+        String realmRoleMappings = (String) roleMappings.get("realm");
+        String testAppScopeMappings = (String) roleMappings.get(diffClient);
+        assertRolesString(realmRoleMappings,
+                "pref.admin",
+                "pref.user",
+                "pref.customer-user-premium"
+        );
+        assertRolesString(testAppScopeMappings,
+                "customer-admin-composite-role",
+                "customer-admin"
+        );
+
+        // Revert
+        deleteMappers(protocolMappers);
```

Comment:
Ensure that this part is executed always, either by leveraging `Creator` in try-with-resources, or via `try-finally` block, or via test cleanup methods. The cleanup also needs to update `directAccessGrant` back to original value.

Reply:
You're right. Thanks @hmlnarik !